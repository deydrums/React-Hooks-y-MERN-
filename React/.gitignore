# proyectos
/00-proyectos