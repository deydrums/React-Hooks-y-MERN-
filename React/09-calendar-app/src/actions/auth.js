import { fetchWithoutToken } from "../helpers/fetch"
import { types } from "../types/types";
import Swal from 'sweetalert2';



export const startLogin = (email, password) => {
    return async(dispatch) => {
        const resp = await fetchWithoutToken('auth',{email,password},'POST');
        const body = await resp.json();
        console.log(body);
        if(resp.ok) {
            localStorage.setItem('token',body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        }else{
            Swal.fire('Error',body.message?body.message:'Ha ocurrido un error','error');
        }
    }
}

export const startRegister = (name, email, password) => {
    return async(dispatch) => {
        const resp = await fetchWithoutToken('auth/new',{name,email,password},'POST');
        const body = await resp.json();
        console.log(body);
        if(resp.ok) {
            localStorage.setItem('token',body.token);
            localStorage.setItem('token-init-date', new Date().getTime());
            dispatch(login({
                uid: body.uid,
                name: body.name
            }));
        }else{
            Swal.fire('Error',body.message?body.message:'Ha ocurrido un error','error');
        }
    }
}

const login = (user) => ({
    type: types.authLogin,
    payload: user
});