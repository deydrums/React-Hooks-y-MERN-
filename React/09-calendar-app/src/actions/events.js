import { fetchWithToken } from "../helpers/fetch";
import { types } from "../types/types";

export const eventStartAddNew = (event) => {
    return async (dispatch, getState) => {

        const {uid, name} = getState().auth;

        try {
            const resp = await fetchWithToken('events', event, 'POST');
            const body = await resp.json();
            if(resp.ok){
                event.id = body.eventDB.id;
                event.user = {
                    _id: uid,
                    name: name
                }
                console.log(event);
                dispatch(eventAddNew(event));
            }
            
        } catch (error) {
            console.log(error);
        }

    }
}


const eventAddNew = (event) => ({
    type: types.eventAddnew,
    payload: event
});

export const eventSetActive = (event) => ({
    type: types.eventSetActive,
    payload: event
});

export const eventUnsetActive = () => ({
    type: types.eventUnsetActive
});

export const eventUpdated = (event) => ({
    type: types.eventUpdated,
    payload: event
});

export const eventDeleted = () => ({
    type: types.eventDeleted
});