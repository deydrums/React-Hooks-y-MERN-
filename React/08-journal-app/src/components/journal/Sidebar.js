import React from 'react';
import { JournalEntries } from './JournalEntries';
import { useDispatch } from 'react-redux';
import { startLogout } from '../../actions/auth';

export const Sidebar = () => {

    const dispatch = useDispatch();

    const handleLogout = () => {
       dispatch(startLogout());
    }
    return (
        <nav className = "journal__sidebar">
            <div className="journal__sidebar-navbar">
                <h3 className = "mt-5">
                    <i className="fa fa-moon"></i>
                    <span> David</span>
                </h3>

                <button 
                    className="btn"
                    onClick = {handleLogout}
                >
                    Cerrar sesiÃ³n
                </button>
            </div>
            <div className="journal__new-entry">
                <i className="far fa-calendar-plus fa-5x"></i>
                <p className = "mt-5">Nueva entrada</p>
            </div>
            <JournalEntries/>
        </nav>
    )
}
