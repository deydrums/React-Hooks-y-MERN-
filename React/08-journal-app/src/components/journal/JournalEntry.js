import React from 'react';
import moment from 'moment';
import { useDispatch } from 'react-redux';
import { activeNote } from '../../actions/notes';

export const JournalEntry = ({id, date, title, body, url}) => {
    const dispatch = useDispatch(); 
    moment.updateLocale('es', {
        months: 'Enero_Febrero_Marzo_Abril_Mayo_Junio_Julio_Agosto_Septiembre_Octubre_Noviembre_Diciembre'.split('_'),
        monthsShort: 'Enero._Feb._Mar_Abr._May_Jun_Jul._Ago_Sept._Oct._Nov._Dec.'.split('_'),
        weekdays: 'Domingo_Lunes_Martes_Miercoles_Jueves_Viernes_Sabado'.split('_'),
        weekdaysShort: 'Dom._Lun._Mar._Mier._Jue._Vier._Sab.'.split('_'),
        weekdaysMin: 'Do_Lu_Ma_Mi_Ju_Vi_Sa'.split('_')
      })
    const noteDate = moment(date);
    const handleEntryClick = () => {
        dispatch(activeNote(id,{
            date,title,body,url
        }));
    }

    return (
        <div className="journal__entry pointer animate__animated animate__fadeIn" onClick={handleEntryClick}>
            {
                url ?
                <div 
                    className="journal__entry-picture"
                    style = {{
                        backgroundSize: 'cover',
                        backgroundImage: `url(${url})`
                    }}
                >
                </div>
                :
                <div 
                className="journal__entry-picture"
                style = {{
                    backgroundSize: 'cover',
                    backgroundColor: 'rgba(102, 250, 193, 0.699)'
                }}
                >
                </div>
            }
            <div className="journal__entry-body">
                <p className="journal__entry-title">
                    {title}
                </p>
                <p className="journal__entry-content">
                    {body}
                </p>
            </div>
            <div className="journal__entry-date-box">
                <span>{noteDate.format('dddd')}</span>
                <h4>{noteDate.format('Do MMM')}</h4>
            </div>
        </div>
    )
}
