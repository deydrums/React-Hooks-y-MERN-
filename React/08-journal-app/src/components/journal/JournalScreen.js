import React from 'react';
import { NoteScreen } from '../notes/NoteScreen';
// import { NothingSelected } from './NothingSelected';
import { Sidebar } from './Sidebar';

export const JournalScreen = () => {
    const btnmoreClick = () => { 
        const q = document.querySelector('.journal__sidebar');
        q.classList.toggle('journal__active');
    }
    const btnmoreOutClick = () => { 
        const q = document.querySelector('.journal__sidebar');
        q.classList.remove('journal__active');
    }
    
    return (
        <div className = "journal__main-content animate__animated animate__fadeIn" >
            <Sidebar/>
            <main onClick = {btnmoreOutClick}>
                {/* <NothingSelected/> */}
                <NoteScreen />
            </main>
            <a 
                className = "journal__btnmore"
                onClick={btnmoreClick}
            >
                <i className="far fa-calendar-plus fa-2x"></i>
            </a>
        </div>
    )
}
