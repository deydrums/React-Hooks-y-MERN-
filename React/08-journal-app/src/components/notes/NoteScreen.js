import React from 'react'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
    return (
        <div className = "notes__main-content">
            <NotesAppBar />
            <div className = "notes__content">
                    <input
                        type = "text"
                        placeholder = "TÃ­tulo..."
                        className = "notes__title-input"
                    />
                    <textarea
                        placeholder = "Que paso hoy?"
                        className = "notes__textarea"
                    >

                    </textarea>

                    <div className = "notes__image">
                        <img 
                            src = "https://64.media.tumblr.com/b38f6615b179cff391fc31456895f6ab/9d110a7eb7405c5c-6f/s2048x3072/f8905f13c648107d1f218b9902f588dfd4ddcb6e.jpg"
                            alt = "image"
                        />
                    </div>

            </div>
        </div>
    )
}
