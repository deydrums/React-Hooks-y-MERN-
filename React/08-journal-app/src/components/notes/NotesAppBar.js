import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startSaveNote, startUploading } from '../../actions/notes';

export const NotesAppBar = () => {
    const dispatch = useDispatch();
    const {active} = useSelector(state => state.notes);
    const handleSave = () => {
        dispatch(startSaveNote(active));
    }

    const handlePictureUpload = () =>{
        document.querySelector('#fileSelector').click();
    }

    const handleFileChange = (e) => {
        const file = e.target.files;
        if(file){
            dispatch(startUploading(file));
        }
    }

    return (
        <div className="notes__appbar">
            <span>28 de agosto 2021</span>
            <input
                type="file"
                style = {{display:'none'}}
                onChange={handleFileChange}
                id = "fileSelector"
                name = "file"
            />

            <div>
                <button className="btn" onClick = {handlePictureUpload}>
                    Fotograf√≠a
                </button>
                <button className="btn" onClick={handleSave}>
                    Guardar
                </button>
            </div>
        </div>
    )
}
