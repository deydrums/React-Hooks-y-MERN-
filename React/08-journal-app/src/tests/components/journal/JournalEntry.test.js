import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

const store = mockStore(initState);
store.dispatch = jest.fn();
const nota = {
    id: 10,
    date: 0,
    title: 'Hola',
    body: 'Mundo',
    url: 'http://deycrums.com/ho.jpg'
};

describe('Pruebas en JournalEntry', () => {
    const wrapper = mount(
        <Provider store={store}>
            <JournalEntry {...nota}/>
        </Provider>
    );

    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de activar la nota', () => {
        wrapper.find('.journal__entry').prop('onClick')();
        expect(store.dispatch).toHaveBeenCalledWith(
            activeNote(nota.id, {...nota})
        );
    });
    
});
