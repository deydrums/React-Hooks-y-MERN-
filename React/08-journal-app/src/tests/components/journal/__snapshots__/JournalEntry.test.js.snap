// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en JournalEntry Debe de mostrarse correctamente 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(http://deycrums.com/ho.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Hola
    </p>
    <p
      className="journal__entry-content"
    >
      Mundo
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Miercoles
    </span>
    <h4>
      31ยบ
    </h4>
    <span>
      Dec.
      1969
    </span>
  </div>
</div>
`;
